### PR TITLE
feat: Adds GCP Secret Manager trigger authentication docs

### PR DESCRIPTION
**Pull request by neelanjan00**
_A description of what has been changed_
- Adds GCP Secret Manager trigger authentication docs

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

### CODE REVIEW

**Review by wozniakjan — COMMENTED** [review]
looks pretty good, just a couple of nits from my side, also the PR title should not be `chore` but `feat`, can you please change that?

**Comment by wozniakjan** [comment]
@neelanjan00 regarding the failing jobs, I think you will need to adjust [`.htmltest`](https://github.com/kedacore/keda-docs/blob/main/.htmltest.yml) like in https://github.com/kedacore/keda-docs/commit/e03ee096b70db4300216c829c6d12c57cff81025

File: content/docs/2.12/authentication-providers/gcp-secret-manager.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
++++
```
[the remark this answers is not in the document]
Comment (zroubalik) [reply content/docs/2.12/authentication-providers/gcp-secret-manager.md:1]: +1

File: content/docs/2.12/authentication-providers/gcp-secret-manager.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+```yaml
+gcpSecretManager:                                       # Optional.
+  gcpProjectID: {gcp-project-id}                      # Required.
+  secrets:                                            # Required.
+    - parameter: {param-name-used-for-auth}           # Required.
+      id: {secret-manager-secret-name}                # Required.
+      version: {secret-manager-secret-name}           # Optional.
+  podIdentity:                                        # Optional.
+    provider: gcp                                     # Required.
+  gcpCredentials:                                     # Optional.
```
Comment (zroubalik) [inline content/docs/2.12/authentication-providers/gcp-secret-manager.md:21]: please rename, https://github.com/kedacore/keda/pull/4864#pullrequestreview-1817786254

**Review by zroubalik — CHANGES_REQUESTED** [review]

**Comment by zroubalik** [comment]
@neelanjan00 please also add this page here to the list, to fix netlify issues: https://github.com/kedacore/keda-docs/blob/02d6f01ba1b8e1aa8b9ce36a18422eef50673bfa/.htmltest.yml#L8

**Comment by neelanjan00** [comment]
Thank you for the detailed reveiw, @wozniakjan @zroubalik. PTAL.

**Review by wozniakjan — APPROVED** [review]
lgtm

**Review by zroubalik — APPROVED** [review]

**Pull request merged**